rest-catalog/quarkus-spark-iceberg-tests/src/intTest/java/org/projectnessie/restcatalog/intttests/ITNessieCatalogServerSparkSQL.java

```diff
@@ -0,0 +1,208 @@
+  public static final String SCOPE = "profile";
+  // Keycloak container
+  static CustomKeycloakContainer keycloakContainer;
+
+  // Nessie Core container
+  static NessieContainer nessieCoreContainer;
+  static String nessieCoreBaseUri;
+
+  // Nessie Catalog
+
+  static String javaExec =
+      requireNonNull(
+          System.getProperty("javaExec"), "Required system property 'javaExec' is not set");
+  static String nessieCatalogServerJar =
+      requireNonNull(
+          System.getProperty("nessie-catalog-server-jar"),
+          "Required system property 'nessie-catalog-server-jar' is not set");
+
+  static final ProcessHandler nessieCatalogProcess = new ProcessHandler();
+  static String nessieCatalogUrl;
+  static String nessieCatalogIcebergUrl;
+
+  @BeforeAll
+  static void startAll() throws Exception {
+    String networkId = Network.SHARED.getId();
+
+    CustomKeycloakContainer.KeycloakConfig config =
+        CustomKeycloakContainer.builder()
+            .dockerNetworkId(networkId)
+            .fromProperties(emptyMap())
+            .realmConfigure(ITNessieCatalogServerSparkSQL::configureRealm)
+            .build();
+
+    keycloakContainer = config.createContainer();
+    keycloakContainer.start();
+
+    NessieContainer.NessieConfig nessieConfig =
+        NessieContainer.builder()
+            .dockerNetworkId(networkId)
+            .fromProperties(emptyMap())
+            .authEnabled(true)
+            .keycloakContainerSupplier(() -> keycloakContainer)
+            .build();
+    nessieCoreContainer = nessieConfig.createContainer();
+    nessieCoreContainer.start();
+    nessieCoreBaseUri = "http://localhost:" + nessieCoreContainer.getExternalNessiePort() + "/api/";
+
+    nessieCatalogProcess.start(
+        new ProcessBuilder(
+            javaExec,
+            "-XX:SelfDestructTimer=30",
+            "-Dquarkus.http.port=0",
+            "-Dnessie.server.authentication.enabled=true",
+            "-Dquarkus.oidc.auth-server-url=" + keycloakContainer.getExternalRealmUri(),
+            "-Dquarkus.oidc.token-issuer=" + keycloakContainer.getTokenIssuerUri(),
+            "-Dquarkus.oidc.client-id=" + serviceClientId,
+            "-Dquarkus.oidc.credentials.secret=" + CLIENT_SECRET,
+            "-Dnessie.iceberg.nessie-client.\"nessie.uri\"=" + nessieCoreBaseUri + "v2/",
+            "-Dnessie.iceberg.nessie-client.\"nessie.authentication.oauth2.client-id\"="
+                + serviceClientId,
+            "-Dnessie.iceberg.nessie-client.\"nessie.authentication.oauth2.client-secret\"="
+                + CLIENT_SECRET,
+            "-jar",
+            nessieCatalogServerJar));
+    nessieCatalogUrl = nessieCatalogProcess.getListenUrl().replace("0.0.0.0", "localhost");
+    nessieCatalogIcebergUrl = nessieCatalogUrl + "/iceberg";
+  }
+
+  private static final String serviceClientId = "quarkus-service-app";
+  private static final String webAppClientId = "quarkus-web-app";
```

Comment:
I think you can skip the creation of this client completely, we don't need webapp clients at all here.